### ISSUE TITLE
Add Polygon HitShape

### ISSUE THREAD

**Issue by reaperrr** [OpenRA/OpenRA#11362]
As mentioned in https://github.com/OpenRA/OpenRA/issues/10349.

```
[15:23] <pchote> you want Exts.PolygonContains
[15:23] <pchote> for polygon hit testing
[15:23] <pchote> I think it would be better to have a polygon hit shape instead of footprint
[15:24] <pchote> because footprints might have holes in them
[15:24] <pchote> and handling that case makes the code at least 5 times more complicated
[15:24] <pchote> the other half of the hitshape is distance from polygon, which you get by taking the minimum distance from each line in the shape
[15:25] <pchote> which i'm pretty sure we have code for
[15:25] <pchote> just need to enumerate the edges
```

**Comment by reaperrr** [comment]
For the record, I tried giving this an attempt, but the math and polygon interaction is way out of my league, I'm not even close to something that qualifies as a starting point. Even with the information provided above, it's not just one or two things I can't handle, it's more like I have question marks over my head for everything except the fact that we need a list of int2 offsets (probably).

Long story short, at this point I can't imagine myself implementing this.

**Comment by reaperrr** [comment]
Closed via #14840.
